static/scripts/PlotHandler.js

```diff
@@ -116,9 +132,87 @@ class PlotWorker {
     }
     return;
   }
+
+  getPlotDataAsCSV(zip) {
+    let csvContent = "";
+    if (this.isSingleInput) {
+      csvContent += singlePlotCalculator.tranformToCSV(this.plotData);
+    } else {
+      csvContent += doublePlotCalculator.tranformToCSV(this.plotData);
+    }
+    zip.file(this.plotName + ".csv", csvContent);
+  }
+}
+
+class singleInputPlotCalculator {
```

Comment:
I just want to know, why do you introduce new classes that don't hold any class variables? :) An idea would be to add a class variable `plotCalculator` with an instance of one of these classes. Then you can use it in getPlotDataAsCSV instead of the if-else call.